src/error.rs

```diff
@@ -1,28 +1,18 @@
 use docopt;
-use failure::{Context, Fail};
-use notion_fail::{ExitCode, NotionError, NotionFail};
-use notion_fail_derive::*;
-
-#[derive(Debug, Fail, NotionFail)]
-#[fail(display = "{}", error)]
-#[notion_fail(code = "InvalidArguments")]
-pub(crate) struct CliParseError {
-    pub(crate) usage: Option<String>,
-    pub(crate) error: String,
-}
-
-impl CliParseError {
-    pub(crate) fn from_docopt(error: &docopt::Error) -> Self {
-        if let &docopt::Error::WithProgramUsage(ref real_error, ref usage) = error {
-            CliParseError {
-                usage: Some(usage.clone()),
-                error: real_error.to_string(),
-            }
-        } else {
-            CliParseError {
-                usage: None,
-                error: error.to_string(),
-            }
+use failure::Context;
+use notion_core::error::ErrorDetails;
+use notion_fail::NotionError;
+
+pub(crate) fn from_docopt_error(error: &docopt::Error) -> ErrorDetails {
```

Comment:
Again I don't think a top-level function should start with `from`. Probably `cli_parse_error` is good.